addons/stock/models/stock_move.py

```diff
@@ -470,9 +470,13 @@ def key_virtual_available(move, incoming=False):
             if not warehouse:  # No prediction possible if no warehouse.
                 continue
             moves = self.browse(moves_ids)
-            forecast_info = moves._get_forecast_availability_outgoing(warehouse)
+            moves_per_location = defaultdict(lambda: self.env['stock.move'])
             for move in moves:
-                move.forecast_availability, move.forecast_expected_date = forecast_info[move]
+                moves_per_location[move.location_id] |= move
+            for location in moves_per_location:
+                forecast_info = moves._get_forecast_availability_outgoing(warehouse, location)
```

Comment:
```suggestion
            for location, mvs in moves_per_location.items():
                forecast_info = mvs._get_forecast_availability_outgoing(warehouse, location)
```
No need to check forecast info on all moves for each location, only those coming from that specific location